scipy/signal/_signaltools.py

```diff
@@ -3540,170 +3566,152 @@ def resample(x, num, t=None, axis=0, window=None, domain='time'):
-    if real_input:
-        newshape[axis] = num // 2 + 1
-    else:
-        newshape[axis] = num
-    Y = np.zeros(newshape, X.dtype)
-
-    # Copy positive frequency components (and Nyquist, if present)
-    N = min(num, Nx)
-    nyq = N // 2 + 1  # Slice index that includes Nyquist if present
-    sl = [slice(None)] * x.ndim
-    sl[axis] = slice(0, nyq)
-    Y[tuple(sl)] = X[tuple(sl)]
-    if not real_input:
-        # Copy negative frequency components
-        if N > 2:  # (slice expression doesn't collapse to empty array)
-            sl[axis] = slice(nyq - N, None)
-            Y[tuple(sl)] = X[tuple(sl)]
-
-    # Split/join Nyquist component(s) if present
-    # So far we have set Y[+N/2]=X[+N/2]
-    if N % 2 == 0:
-        if num < Nx:  # downsampling
-            if real_input:
-                sl[axis] = slice(N//2, N//2 + 1)
-                Y[tuple(sl)] *= 2.
-            else:
-                # select the component of Y at frequency +N/2,
-                # add the component of X at -N/2
-                sl[axis] = slice(-N//2, -N//2 + 1)
-                Y[tuple(sl)] += X[tuple(sl)]
-        elif Nx < num:  # upsampling
-            # select the component at frequency +N/2 and halve it
-            sl[axis] = slice(N//2, N//2 + 1)
-            Y[tuple(sl)] *= 0.5
-            if not real_input:
-                temp = Y[tuple(sl)]
-                # set the component at -N/2 equal to the component at +N/2
-                sl[axis] = slice(num-N//2, num-N//2 + 1)
-                Y[tuple(sl)] = temp
-
-    # Inverse transform
-    if real_input:
-        y = sp_fft.irfft(Y, num, axis=axis)
-    else:
-        y = sp_fft.ifft(Y, axis=axis, overwrite_x=True)
-
-    y *= (float(num) / float(Nx))
-
-    if t is None:
-        return y
+    xp = array_namespace(x, t)
+    x = xp.asarray(x)
+    if x.ndim > 1:  # moving active axis to end allows to use `...` in indexing:
+        x = xp.moveaxis(x, axis, -1)
+    n_x = x.shape[-1]  # number of samples along the time/frequency axis
+    s_fac = n_x / num  # scaling factor represents sample interval dilatation
+    m = min(num, n_x)  # number of relevant frequency bins
+    m2 = m // 2 + 1  # number of relevant frequency bins of a one-sided FFT
+
+    if window is None: # Determine spectral windowing function:
+        W = None
+    elif callable(window):
+        W = window(sp_fft.fftfreq(n_x))
+    elif hasattr(window, 'shape'): # must be an array object
+        if window.shape != (n_x,):
+            raise ValueError(f"{window.shape=} != ({n_x},), i.e., window length " +
+                             "is not equal to number of frequency bins!")
+        W = xp.asarray(window, copy=True)  # prevent modifying the function parameters
     else:
-        new_t = np.arange(0, num) * (t[1] - t[0]) * Nx / float(num) + t[0]
-        return y, new_t
+        W = sp_fft.fftshift(get_window(window, n_x))
+
+    if domain == 'time' and xp.isdtype(x.dtype, 'real floating'):  # utilize rfft():
```

Comment:
So integer signals go through the two-sided FFT code path, even in time domain, is this intentional?

Reply:
Good catch! Addressed in commit 76636f9e7.